File: backend/src/main/java/com/football/backend/dto/EditionDashboardDto.java
package com.football.backend.dto;

import java.util.List;
import java.util.UUID;

public class EditionDashboardDto {
    UUID editionId;
    UUID competitionId;
    String editionName;
    String competitionName;
    List<StandingsEntryDto> table;
    List<List<MatchListDto>> rounds;
}

File: frontend/src/dto/EditionDashboardDto.ts
export interface MatchListDto {
    id: string;
    homeTeamName: string;
    awayTeamName: string;
    homeGoals: number;
    awayGoals: number;
    matchDate: string;
    status: string;
}

export interface StandingsEntryDto {
    teamId: string;
    teamName: string;
    poIntegers: number;
    played: number;
    wins: number;
    draws: number;
    losses: number;
    goalsFor: number;
    goalsAgainst: number;
    goalDifference: number;
}

export interface EditionDashboardDto {
    editionId: string;
    competitionId: string;
    editionName: string;
    competitionName: string;
    table: StandingsEntryDto[]; // Empty if Knockout
    rounds: MatchListDto[][];
}